### ISSUE TITLE
Error Reading File: /dev/stdin

### ISSUE THREAD

**Issue by ZevEisenberg** [dgyesbreghs/install_provisioning_profiles#4]
I've got an error while running version 0.2.2 of this plugin inside of fastlane 2.115.0. I've narrowed it down to [this line](https://github.com/dgyesbreghs/install_provisioning_profiles/blob/master/lib/fastlane/plugin/install_provisioning_profiles/actions/install_provisioning_profiles_action.rb#L52). I can reproduce the issue in my Terminal by running this command:

```sh
/usr/libexec/PlistBuddy -c 'Print :UUID' /dev/stdin <<< $(security cms -D -I path/to/my/profile 2> /dev/null)
```

This prints out the following:

```
Error Reading File: /dev/stdin
```

When I run just the middle part, the part inside the `$(...)`, it produces valid XML output. So I think there's something wrong with the `/dev/stdin <<<` part. I don't fully understand that part of the shell syntax, though, so I could be wrong.

Has anyone else seen this error? Any ideas about how to fix it?

**Comment by AliSoftware** [comment]
* `<<< x` in shell means "feed the text 'x' to stdin"
* `$(…)` is similar to putting the same expression in backticks: it executes the inner expression and replace it with the returned value

So what `<<< $(security …)` does overall is to execute the command `security …` in a sub-shell, then use the result of that command to feed stdin. Finally, you're telling PlistBuddy to read stdin.

I'm not close to a Mac to test it but I think that this could be rewritten using the `<(…)` syntax instead, like this:
```
/usr/libexec/PlistBuddy -c 'Print :UUID' <(security cms -D -I path/to/my/profile 2>/dev/null)
```
As the shell syntax `<(…)` has the effect of executing the inner command and make the result appear as if being the content of a file you'd pass to the main command, so exactly the semantic that `/dev/stdin` + `<<< $(…)` seem to try to achieve here.

**Comment by AliSoftware** [comment]
Also I'm not sure if that's the case for PlistBuddy but some commands assume stdin when not given an input file; and other commands accept `-` for the input file parameter to mean "read from stdin". So if PlistBuddy is in one of those two cases, another alternative would be to use a simple pipe: `security … | /usr/libexec/PlistBuddy -c 'Print :UUID' -`. You'll have to test on a Mac to see if PlistBuddy accepts such `-` pseudo-filename as parameter, or otherwise test without that `-` hopping that it defaults to stdin (on my phone rn so can't test it myself)

**Comment by ZevEisenberg** [comment]
I believe I read online that PlistBuddy doesn't use `-` as stdin, but `plutil` does. But I believe this code used to work, so I'm wondering what changed.

**Comment by AliSoftware** [comment]
Ah right I might have thought of plutil when in thought about `-` indeed.

In any way, using `<(…)` instead of `/dev/stdin <<< $(…)` should work (and might even be more efficient in term of redirection mechanisms and number of sub shells being created) and won't involve /dev/stdin so will likely solve the issue.

**Comment by ZevEisenberg** [comment]
```
$ /usr/libexec/PlistBuddy -c 'Print :UUID' <(security cms -D -i /path/to/my/profile 2> /dev/null)
Error Reading File: /dev/fd/11
```

**Comment by AliSoftware** [comment]
Wow ok that means there's a deeper underlying problem then, it's not just unable to read `/dev/stdin` but any file / file descriptor?

**Comment by ZevEisenberg** [comment]
¯\\\_(ツ)\_/¯ it works when I write out the results of `security cms -D -i` to a file, and then read it back in with PlistBuddy. Maybe the script should just do that. I'm just confused because this script has been working for years, as far as I can tell. Maybe something broke in Mojave?

**Comment by AliSoftware** [comment]
I'm experimenting the same with High Sierra here.

fwiw, using `plutil` seems to lead to similar errors anyway:
```bash
$ plutil -p <(security cms -D -i 5a924f53-7161-4212-a5df-e304ce5c4451.mobileprovision 2>/dev/null)
/dev/fd/11: file does not exist or is not readable or is not a regular file (Error Domain=NSCocoaErrorDomain Code=257 "The file “11” couldn’t be opened because you don’t have permission to view it." UserInfo={NSFilePath=/dev/fd/11, NSUnderlyingError=0x7ff228608350 {Error Domain=NSPOSIXErrorDomain Code=13 "Permission denied"}})
```

**Comment by mr-fixit** [comment]
PlistBuddy isn't a well-behaved *nix CLI tool, if it were, it would read `stdin`.

If you don't mind having more than 1 command, then do `TMP=$(mktemp); security cms ... > $TMP; PListBuddy $TMP; rm $TMP`

If you really want 1 command, and don't mind a domain-specific grep-based XML parsing hack, `security cms -D -i $PROFILE | grep -A 1 '<key>UUID</key>' | tail -n 1 | grep -Eo '>.*<' | grep -Eo '[a-z0-9-]+`

**Comment by dgyesbreghs** [comment]
I have just released a new version (0.3.0). Sorry for the delay.
